homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,213 @@
+
+        try:
+            data = await self.gti.departureList(payload)
+        except InvalidAuth as error:
+            if self._last_error != InvalidAuth:
+                _LOGGER.error("Authentication failed: %r", error)
+                self._last_error = InvalidAuth
+            self._available = False
+        except ClientConnectorError as error:
+            if self._last_error != ClientConnectorError:
+                _LOGGER.warning("Network unavailable: %r", error)
+                self._last_error = ClientConnectorError
+            self._available = False
+        except Exception as error:  # pylint: disable=broad-except
+            if self._last_error != error:
+                _LOGGER.error("Error occurred while fetching data: %r", error)
+                self._last_error = error
+            self._available = False
+
+        if data["returnCode"] == "OK" and data.get("departures"):
+
+            if self._last_error == ClientConnectorError:
+                _LOGGER.debug("Network available again")
+
+            self._last_error = None
+
+            departure = data["departures"][0]
+            self._available = True
+            self._state = (
+                departure_time
+                + timedelta(minutes=departure["timeOffset"])
+                + timedelta(seconds=departure.get("delay", 0))
+            )
+
+            self.attr[ATTR_LINE] = departure["line"]["name"]
+            self.attr[ATTR_ORIGIN] = departure["line"]["origin"]
+            self.attr[ATTR_DIRECTION] = departure["line"]["direction"]
+            self.attr[ATTR_TYPE] = departure["line"]["type"]["shortInfo"]
+            self.attr[ATTR_ID] = departure["line"]["id"]
+            self.attr[ATTR_DELAY] = departure.get("delay", 0)
+
+            departures = []
+            for departure in data["departures"]:
+                departures.append(
+                    {
+                        ATTR_DEPARTURE: departure_time
+                        + timedelta(minutes=departure["timeOffset"])
+                        + timedelta(seconds=departure.get("delay", 0)),
+                        ATTR_LINE: departure["line"]["name"],
+                        ATTR_ORIGIN: departure["line"]["origin"],
+                        ATTR_DIRECTION: departure["line"]["direction"],
+                        ATTR_TYPE: departure["line"]["type"]["shortInfo"],
+                        ATTR_ID: departure["line"]["id"],
+                        ATTR_DELAY: departure.get("delay", 0),
+                    }
+                )
+            self.attr[ATTR_NEXT] = departures
+        else:
+            self._available = False
+
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this sensor."""
+        station_id = self.config_entry.data["station"]["id"]
+        station_type = self.config_entry.data["station"]["type"]
+
+        return f"{DOMAIN}-{self.config_entry.entry_id}-{station_id}-{station_type}"
```

Comment:
You can remove DOMAIN since unique_id only need to be unique for your integration.
```suggestion
        return f"{self.config_entry.entry_id}-{station_id}-{station_type}"
```

Reply:
I plan to add other platforms (binary_sensor) which share the same station_id and station_type, that's why I added the domain. Under these circumstances, can I keep the domain in the UUID?

Reply:
No, you can drop the domain in unique_ids because:

> An entity is looked up in the registry based on a combination of the plaform type (e.g., light), and the integration name (domain) (ie hue) and the unique ID of the entity.

source: https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements